src/pages/index.js

```diff
@@ -89,31 +134,105 @@ function handleImageClick(data) {
 /*Event Handler*/
 
 function handleProfileEditSubmit(inputValue) {
+  api.updateProfile({
+    name: inputValue.title,
+    description: inputValue.description
+  })
+  .then(updatedUserData => {
     userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+      name: updatedUserData.name,
+      about: updatedUserData.about,
     });
     popupWithEditProfileForm.close();
+  })
+  .catch(err => {
+    console.error(err);
+  });
 }
 
 function handleAddCardFormSubmit(inputValue) {
     const cardData = {
         name: inputValue.title,
         link: inputValue.url,
     };
-    renderCard(cardData);
     popupWithAddCardForm.close();
     addCardForm.reset();
```

Comment:
```js

    popupWithAddCardForm.close();
    addCardForm.reset();
```

They should be in `then` in line 162

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**